pandas/core/groupby/ops.py

```diff
@@ -258,7 +258,12 @@ def groups(self):
         if len(self.groupings) == 1:
             return self.groupings[0].groups
         else:
-            to_groupby = zip(*(ping.grouper for ping in self.groupings))
+            def is_basegrouper(self, obj):
+                return obj.__class__.__name__ == self.__class__.__name__
+            to_groupby = zip(*(ping.grouper if not is_basegrouper(self,
+                                                                  ping.grouper)
+                             else ping.grouper.groupings[0].grouper for ping
+                             in self.groupings))
```

Comment:
here's a patch to solve this in a cleaner way; the issue is that there can be an embeded grouper in construction; but we always convert to an Index for actual grouping (except for the BinGroupers).

```
(pandas) bash-3.2$ git diff
diff --git a/pandas/core/groupby/grouper.py b/pandas/core/groupby/grouper.py
index 63931dda6..d2ee38a74 100644
--- a/pandas/core/groupby/grouper.py
+++ b/pandas/core/groupby/grouper.py
@@ -256,7 +256,6 @@ class Grouping:
 
         # we have a single grouper which may be a myriad of things,
         # some of which are dependent on the passing in level
-
         if level is not None:
             if not isinstance(level, int):
                 if level not in index.names:
@@ -280,7 +279,7 @@ class Grouping:
             if self.name is None:
                 self.name = grouper.result_index.name
             self.obj = self.grouper.obj
-            self.grouper = grouper
+            self.grouper = grouper._get_grouper()
 
         else:
             if self.grouper is None and self.name is not None:
diff --git a/pandas/core/groupby/ops.py b/pandas/core/groupby/ops.py
index 235975a9b..a031a25f0 100644
--- a/pandas/core/groupby/ops.py
+++ b/pandas/core/groupby/ops.py
@@ -150,6 +150,15 @@ class BaseGrouper:
         comp_ids, _, ngroups = self.group_info
         return get_splitter(data, comp_ids, ngroups, axis=axis)
 
+    def _get_grouper(self):
+        """
+        We are a grouper as part of another's groupings
+
+        We are a simpler grouper, who can
+        convert to a Index for our grouper.
+        """
+        return self.groupings[0].grouper
+
     def _get_group_keys(self):
         if len(self.groupings) == 1:
             return self.levels[0]
@@ -707,6 +716,15 @@ class BinGrouper(BaseGrouper):
     def nkeys(self):
         return 1
 
+    def _get_grouper(self):
+        """
+        We are a grouper as part of another's groupings.
+
+        We have a specific method of grouping, so cannot
+        convert to a Index for our grouper.
+        """
+        return self
+
     def get_iterator(self, data, axis=0):
         """
         Groupby iterator
```

Reply:
This is indeed much cleaner, will patch this in the next commit.